src/components/TreeVisualizer.jsx

```
@@ -978,12 +1257,39 @@ const TreeVisualizer = () => {
+        animationFrame.current = null;
+      }
+      
+      // Ensure we preserve the final tree state if animation was in progress
+      const { results, index } = animationState.current;
+      if (results && results.length > 0 && index > 0) {
+        // Get the last processed animation step
+        const lastProcessedIndex = Math.min(index, results.length - 1);
+        const lastResult = results[lastProcessedIndex];
+        
+        // Update tree to this state if it has a tree property
+        if (lastResult && lastResult.hasOwnProperty('tree')) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)